### ISSUE TITLE
v0.1.0 Planning & Discussions

### ISSUE THREAD

**Issue by romans127** [romans127/vscode-svn#1]
Planning & Discussions for v0.1.0 Release

Below are a few things that will need to be ironed out in order to get the ball rolling on this project. 

- [ ] Model / Replicate `scm-hg` or `scm-git` ? *this depends on which implementation is closer related to how we will need to structure `scm-svn`*
- [ ] Use cli or node calls to svn vs. opening a prompt for TortoiseSVN , like the other implementations in the marketplace. `spawn-svn` proposed
- [ ] Determine a list of features to start with , for `v0.1.0` then create an iteration plan to roll out the other features.
- [ ] Determine a list of features that "can" be supported via the scm provider api.

**Comment by csholmq** [comment]
Not too familiar with either Mercury (Hg) nor Git. But from what I understand, Hg is fairly simple and to the point. So perhaps less fuss?

What would be the benefits of using TortoiseSVN? Speed of implementation and familiar GUI? I say we go for CLI and aim to match the experience of Git and Hg. Built in diffs, icons, etc.

I propose we start with supporting the basic commit, upgrade, add and delete. This should include indicating when files are uncommitted, added, etc. Then add branch and tag functionalities, diffing, log viewing, etc.

**Comment by s-martin** [comment]
I'd like to propose not to rely on TortoiseSVN, because it's not platform independent and I would use a svn extension both on Windows and Linux.

There are a couple of svn implementations available for node.js (e.g. spawn-svn), but I'm not familiar which one would suit best.

**Comment by romans127** [comment]
I agree I want to have the workflow very inline with the existing git scm

**Comment by romans127** [comment]
Just posting an update. 
Now that i've received some feedback, I will probably start the first round of development next week.

**Comment by s-martin** [comment]
+1

**Comment by gruberr** [comment]
+1 for a cross platform solution that doesn't require specific versions of SVN installed. Would like to contribute/fork if time and code structure permits... currently this cindarella like SVN support is the only thing holding me back from doing more work "cross platform" with minimal effort.

**Comment by cwrau** [comment]
👍  for platform independent solution

I would like to contribute aswell, I can't wait for SVN support 😃

**Comment by s-martin** [comment]
I'd be willing to support and contribute despite my limited typescript experience.

**Comment by magic-k** [comment]
Hi, has there been any progress?
